File: app/src/main/java/com/example/orderingapp/Adapter/OrderItemAdapter.java
package com.example.orderingapp.Adapter;


import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.example.orderingapp.Domain.OrderItem;
import com.example.orderingapp.R;


import java.util.List;

public class OrderItemAdapter extends RecyclerView.Adapter<OrderItemAdapter.OrderViewHolder> {

    private List<OrderItem> orderList;

    public OrderItemAdapter(List<OrderItem> orderList) {
        this.orderList = orderList;
    }

    public static class OrderViewHolder extends RecyclerView.ViewHolder {
        ImageView itemImage;
        TextView itemName, itemPrice, itemQuantity;

        public OrderViewHolder(View itemView) {
            super(itemView);
            itemImage = itemView.findViewById(R.id.item_image);
            itemName = itemView.findViewById(R.id.item_name);
            itemPrice = itemView.findViewById(R.id.item_price);
            itemQuantity = itemView.findViewById(R.id.item_quantity);
        }
    }

    @Override
    public OrderViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_order_summary, parent, false);
        return new OrderViewHolder(view);
    }

    @Override
    public void onBindViewHolder(OrderViewHolder holder, int position) {
        OrderItem orderItem = orderList.get(position);

        // Change getName() to getTitle() if your field is 'title'
        holder.itemName.setText(orderItem.getTitle());

        holder.itemPrice.setText("$" + String.format("%.2f", orderItem.getPrice()));
        holder.itemQuantity.setText(String.valueOf(orderItem.getQuantity()));

        String imageUrl = orderItem.getPicUrl();
        if (imageUrl != null && !imageUrl.isEmpty()) {
            Glide.with(holder.itemView.getContext())
                    .load(imageUrl)
                    .placeholder(R.drawable.capybara)
                    .error(R.drawable.capybara)
                    .into(holder.itemImage);
        } else {
            holder.itemImage.setImageResource(R.drawable.capybara);
        }
    }

    @Override
    public int getItemCount() {
        return orderList.size();
    }
}

File: app/src/main/java/com/example/orderingapp/Adapta/OrderItemAdapter.kt
package com.example.orderingapp.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.orderingapp.Domain.OrderItem
import com.example.orderingapp.R

class OrderItemAdapter(
    private val orderList: List<OrderItem>
) : RecyclerView.Adapter<OrderItemAdapter.OrderViewHolder>() {

    inner class OrderViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val itemImage: ImageView = itemView.findViewById(R.id.item_image)
        val itemName: TextView = itemView.findViewById(R.id.item_name)
        val itemPrice: TextView = itemView.findViewById(R.id.item_price)
        val itemQuantity: TextView = itemView.findViewById(R.id.item_quantity)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): OrderViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_order_summary, parent, false)  // Your item layout file
        return OrderViewHolder(view)
    }

    override fun onBindViewHolder(holder: OrderViewHolder, position: Int) {
        val orderItem = orderList[position]

        holder.itemName.text = orderItem.name
        holder.itemPrice.text = "$${String.format("%.2f", orderItem.price)}"
        holder.itemQuantity.text = orderItem.quantity.toString()

        // Set image - for now using placeholder drawable
        holder.itemImage.setImageResource(R.drawable.capybara)

        // TODO: Use Glide or Picasso if you have image URLs
    }

    override fun getItemCount(): Int = orderList.size
}
